astropy/modeling/core.py

```diff
@@ -2229,8 +2236,8 @@ def _prepare_outputs_single_model(self, outputs, broadcasted_shapes):
         outputs = list(outputs)
         for idx, output in enumerate(outputs):
             try:
-                broadcast_shape = check_broadcast(*broadcasted_shapes[0])
-            except (IndexError, TypeError):
+                broadcast_shape = np.broadcast_shapes(*broadcasted_shapes[0])
+            except Exception:
```

Comment:
note that I made the error catching *less* accurate here so it would also capture `DeprecationWarning`s when they're elevated to errors in testing, and which should become actual exceptions of unknown types in the future.